lib/active_fedora/file/streaming.rb

```diff
@@ -22,13 +22,24 @@ def initialize(uri, headers)
       @headers = headers
     end
 
-    def each
+    def each(no_of_requests_limit = 3)
+      raise ArgumentError, 'HTTP redirect too deep' if no_of_requests_limit == 0
       Net::HTTP.start(uri.host, uri.port) do |http|
         request = Net::HTTP::Get.new uri, headers
         http.request request do |response|
-          raise "Couldn't get data from Fedora (#{uri}). Response: #{response.code}" unless response.is_a?(Net::HTTPSuccess)
-          response.read_body do |chunk|
-            yield chunk
+          case response
+          when Net::HTTPSuccess
+            response.read_body do |chunk|
+              yield chunk
+            end
+          when Net::HTTPRedirection
+            no_of_requests_limit -= 1
+            @uri = URI(response["location"])
+            each no_of_requests_limit do |chunk|
```

Comment:
If you change the signature of `each` to take a `&block`, you can just pass the block back to `each` like this: `each(no_of_requests_limit, &block)`